inference/cluster_store.py

```diff
@@ -24,12 +97,88 @@ def average_similarity(self):
         # TODO 
         pass
 
+    def euclidian_dist(self,entity1, entity2):
+        '''
+        :param entity1: X
+        :param entity2: Y
+        :return: euclidian distance between two entity vector (3D) X and Y
+        '''
+        X = self.original_data[entity1]
+        Y = self.original_data[entity2]
+        return math.sqrt(pow((X[0]-Y[0]),2) + pow((X[1]-Y[1]),2) + pow((X[2]-Y[2]),2))
+
+    def avg_distance_between_two_cluster(self,k1,k2):
+        '''
+        :param k1: cluster1
+        :param k2: cluster2
+        :return: average distance between two cluster
+        :formula: dist(Ki,Kj) = mean(dist(eil,ejm)) where every eil belongs to Ki not belongs to Kj
+                                                    and every ejk belongs to Kj not belongs to Ki
+        '''
+        dist = []
+        for k1_entity in k1:
+            for k2_entity in k2:
+                dist.append(self.euclidian_dist(k1_entity,k2_entity))
+        return statistics.mean(dist)
+
     # Function to calculate average distance of current clusters 
     def average_distance(self):
-        # TODO
-        pass
+        '''
+        :return: average distance of all clusters store in avg_distance
+        '''
+        dist = []
+        for i in range(self.total_cluster):
+            for j in range(i+1, self.total_cluster):
+                dist.append(self.avg_distance_between_two_cluster(self.clusters[i],self.clusters[j]))
+        self.avg_distance = statistics.mean(dist)
+
+    # map cluster with entity name and return the clusters
+    def get_clusters_on_entity_map(self):
+        return_ = self.clusters
+        for cluster in range(self.total_cluster):
+            entitys = [self.entity_map[x] for x in self.clusters[cluster]]
+            return_[cluster]= entitys
+            # return_[cluster].replace()
+        return return_
 
+    # map cluster with data and return the clusters
+    def get_clusters_on_data(self):
```

Comment:
Seems like a duplicate function. I'm assuming this will be removed?

Reply:
I was little bit confused about it. that's why i keep these two getter options. Okay i'm going to delete this one.